common/src/main/java/com/wynntils/features/combat/RangeVisualizerFeature.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.combat;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.features.Feature;
+import com.wynntils.mc.event.PlayerRenderEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.models.gear.type.GearInfo;
+import com.wynntils.models.gear.type.GearMajorId;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.buffered.CustomRenderType;
+import net.minecraft.client.gui.screens.inventory.InventoryScreen;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.core.Position;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.joml.Matrix4f;
+
+@ConfigCategory(Category.COMBAT)
+public class RangeVisualizerFeature extends Feature {
+
+    private static final MultiBufferSource.BufferSource BUFFER_SOURCE =
+            MultiBufferSource.immediate(new BufferBuilder(256));
+    private static final int SEGMENTS =
+            128; // number of straight lines to draw when rendering circle, higher = smoother but more expensive
+    private static final float HEIGHT = 0.1f;
+
+    @SubscribeEvent
+    public void onPlayerRender(PlayerRenderEvent e) {
+        if (!Models.Player.isLocalPlayer(e.getPlayer())) return; // Don't render for ghost/npc
```

Comment:
These are a lot of checks to make on each render. I worry it might be too expensive. Maybe you should do these checks once per tick instead, and just store radius and color for the rendering code to pick up and just render the circle from?